### DISCUSSION TITLE
Comfortable (offline) multiple map activation & other map related things

### DISCUSSION THREAD

**Discussion by DIrgi** [devemux86/cruiser#519] [Q&A] [answered]
Switching between maps as well online/offline always end in a deactivation of the last map(s). So far so good. 
Doing this while remaining in the same mode (online or offline) is more or less clear.
Switching between online and offline maps already not that clear anymore.

Example: When I activated several maps in (offline) Mapforge, switching to an online map, going back once, all offline Mapforge maps are deactivated again - Why they cannot be automatically reactivated based on their last selection - I assume again a a greater  programming effort?

But if so - Will it then be possible to (re)activate all existing Mapforce maps (additionally) at once by "one click"? - 
So far I have to activate all of them in a single way.  
There can be two more ways next to the existing single map by map activation, 
either "activate all" by "mark them all" or doing fast clicks on each single map and then finally 
"activate all selected maps".

Would make map activations more comfortable.

**Comment by devemux86** [comment]
@DIrgi thanks for the suggestion.

Offline maps will certainly have more improvements in the future.
We can collect everyone's ideas and see what is possible and can be implemented.

Specifically their management list will have a new implementation.

**Comment by devemux86** [comment]
@DIrgi and everyone else,

When you see things that need improvement, please create new topics and write them so we can discuss them. :slightly_smiling_face:

**Comment by DIrgi** [comment]
### **Proposal 1:**
- Keeping map selections

In general it's good to keep map selections when you switch between map. Depending on the area sometimes I prefer rather this or that map.
It would be always great when you switch back to a Map i.e. from Freizeitkarte to Mapforge the maps there you activated once will be reactivated again.

**Comment by DIrgi** [comment]
### **Proposal 2:**
-Switching map color automatically between day/night

Normally every modern device - smartphone, tablet has a light sensor and if set on auto, the screen will be more or less bright according to the environment brightness..but the map color or design itself remains same.
In settings we have additionally the map color, where we can decide manually switching between different map appearances.

Can Cruiser monitor the internal light sensor and switch between day and night (on demand) 
on - by user pre selected map colors?
So i.e. on daylight I select "high contrast" when it becomes dark outside I pre-selected "dark" and Cruiser does the switching according to the light sensor.

**Reply by MotoUKRider** [reply]
@DIrgi That was [one of my earlier requests](https://github.com/devemux86/cruiser/discussions/126) :smile: 
That's one of the reasons I specifically created the "Dark" theme, to future-proof any day/night switching within the app.

**Reply by DIrgi** [reply]
But well, your "Motorider" design can be changed by the map color in the settings. There was no necessity to develop2 design themes.
Or was it at these days not existing - The map color possibility?

**Reply by devemux86** [reply]
> Can Cruiser monitor the internal light sensor and switch between day and night (on demand)

Auto day / night map is planned, there several things to check. We can discuss it in its topic:

- https://github.com/devemux86/cruiser/discussions/126

**Comment by MotoUKRider** [comment]
> But well, your "Motorider" design can be changed by the map color in the settings. There was no necessity to develop2 design themes. Or was it at these days not existing - The map color possibility?

The app's Map Color setting does not paint the map truly "dark", try it - most of the background is a light grey. The "Dark" theme specifically keeps everything but your location+route as low-profile as possible and makes a big difference whether in the car or on the bike.

**Comment by devemux86** [comment]
Next Cruiser will have a redesigned offline map manager with multiple selection, activation, deletion, etc.

<img src="https://github.com/user-attachments/assets/e0e09127-c35d-4b53-b1d9-5a07c80bdff6" width="50%"/><img src="https://github.com/user-attachments/assets/9ef4d674-cd3d-4fd7-8b8b-2576d97d6751" width="50%"/>

**Comment by devemux86** [comment, marked as answer]
- A redesigned offline map manager with multiple selection, activation, deletion, etc. is available in https://github.com/devemux86/cruiser/discussions/641
